cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -282,27 +298,8 @@ public String getValue() {
 
   public static Collection<LoggingContext.SystemTag> getTags(final LoggingContext loggingContext) {
     final List<String> tagNames = new ArrayList<>();
-    if (loggingContext instanceof ApplicationLoggingContext) {
-      tagNames.add(ApplicationLoggingContext.TAG_RUN_ID);
-      tagNames.add(ApplicationLoggingContext.TAG_INSTANCE_ID);
-    }
-
-    if (loggingContext instanceof FlowletLoggingContext) {
-      tagNames.add(FlowletLoggingContext.TAG_FLOW_ID);
-      tagNames.add(FlowletLoggingContext.TAG_FLOWLET_ID);
-    } else if (loggingContext instanceof WorkflowLoggingContext) {
-      tagNames.add(WorkflowLoggingContext.TAG_WORKFLOW_ID);
-    } else if (loggingContext instanceof MapReduceLoggingContext) {
-      tagNames.add(MapReduceLoggingContext.TAG_MAP_REDUCE_JOB_ID);
-    } else if (loggingContext instanceof SparkLoggingContext) {
-      tagNames.add(SparkLoggingContext.TAG_SPARK_JOB_ID);
-    } else if (loggingContext instanceof UserServiceLoggingContext) {
-      tagNames.add(UserServiceLoggingContext.TAG_USER_SERVICE_ID);
-      tagNames.add(UserServiceLoggingContext.TAG_HANDLER_ID);
-    } else if (loggingContext instanceof WorkerLoggingContext) {
-      tagNames.add(WorkerLoggingContext.TAG_WORKER_ID);
-    } else {
-      throw new IllegalArgumentException(String.format("Invalid logging context: %s", loggingContext));
+    for (LoggingContext.SystemTag systemTag : loggingContext.getSystemTags()) {
+      tagNames.add(systemTag.getName());
```

Comment:
this method seems unnecessary, looks like this method is just creating a copy of system tags in loggingContext, do we need this?